packages/blocks/src/api/raw-handling/utils.js

```diff
@@ -37,18 +37,35 @@ export function getBlockContentSchema( transforms ) {
 				schema[ tag ].attributes.push( 'id' );
 			}
 		}
+		if ( isMatch ) {
+			for ( const tag in schema ) {
+				schema[ tag ].isMatch = isMatch;
+			}
+		}
 		return schema;
 	} );
 
 	return mergeWith( {}, ...schemas, ( objValue, srcValue, key ) => {
-		if ( key === 'children' ) {
-			if ( objValue === '*' || srcValue === '*' ) {
-				return '*';
-			}
+		switch ( key ) {
+			case 'children': {
+				if ( objValue === '*' || srcValue === '*' ) {
+					return '*';
+				}
 
-			return { ...objValue, ...srcValue };
-		} else if ( key === 'attributes' || key === 'require' ) {
-			return [ ...( objValue || [] ), ...( srcValue || [] ) ];
+				return { ...objValue, ...srcValue };
+			}
+			case 'attributes':
+			case 'require': {
+				return [ ...( objValue || [] ), ...( srcValue || [] ) ];
+			}
+			case 'isMatch': {
+				if ( ! objValue || ! srcValue ) {
+					return undefined;
+				}
+				return ( ...args ) => {
```

Comment:
Would it be possible to comment on this? It's hard to follow this if right away.

Reply:
A description of the overall flow of the `mergeWith` call would help too.